lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,226 @@
+              end
+            end
+          end
+        end
+
+        def get_text_content_url(url)
+          # Convert github file URLs to raw URLs
+          if match = url.match(/https?:\/\/(?:www\.)?github.com\/([^\/]+)\/([^\/]+)\/blob\/(.*)/i)
+            url = "https://github.com/#{match[1]}/#{match[2]}/raw/#{match[3]}"
+          else
+            url
+          end
+        end
+
+        def inspect_url_for_license(url)
+          if match = url.match(/https?:\/\/licenses.nuget.org\/(.*)/i)
+            Licensee::License.find(match[1])
+          elsif match = url.match(/https?:\/\/opensource.org\/licenses\/(.*)/i)
+            Licensee::License.find(match[1])
+          elsif match = url.match(/https?:\/\/(?:www\.)?apache.org\/licenses\/(.*?)(?:\.html|\.txt)?$/i)
+            Licensee::License.find(match[1].gsub("LICENSE", "Apache"))
+          end
+        end
+
+        def ignored_url?(url)
+          url == "https://aka.ms/deprecateLicenseUrl"
+        end
+
+        def retrieve_license(url, redirect_limit = 2)
+          return if redirect_limit == 0
+          begin
+            response = Net::HTTP.get_response(URI(url))
+            case response
+            when Net::HTTPSuccess     then response.body
+            when Net::HTTPRedirection then retrieve_license(response["location"], redirect_limit - 1)
+            end
+          rescue
+          end
+        end
+      end
+
+      def enabled?
+        nuget_config_exists(config.pwd)
+      end
+
+      def nuget_obj_root
+        config.dig("nuget", "obj_root") || config.pwd
+      end
+
+      def nuget_config_exists(root)
+        # Multiple supported casings: https://github.com/NuGet/Home/issues/1427
+        File.exist?(File.join(root, "nuget.config")) || File.exist?(File.join(root, "NuGet.config")) || File.exist?(File.join(root, "NuGet.Config"))
+      end
+
+      def enumerate_dependencies
+        packages.map do |key, package|
+          NuGetDependency.new(
+            name: "#{package["name"]} #{package["version"]}",
+            version: package["version"],
+            path: package["path"],
+            metadata: {
+              "type"     => NuGet.type,
+              "name"     => package["name"]
+            }
+          )
+        end
+      end
+
+      # Inspect project.assets.json files for package references.
+      # Ideally we'd use `dotnet list package` instead, but its output isn't
+      # easily machine readable, and it also requires repos to have a .sln file
+      # to evaluate multiple projects.
+      def packages
+        return @packages_by_id if defined?(@packages_by_id)
+        @packages_by_id = Hash.new
+        Dir.glob(File.join(nuget_obj_root, "**/project.assets.json")).map do |file|
+          gather_packages(file)
+        end
+        @packages_by_id
+      end
+
+      def gather_packages(project_assets_file)
```

Comment:
Is it expected to pull in transitive dependencies? I assumed so, but just wanted to check since it results in lots of .NET Core implementation detail packages that have to be ignored.

Reply:
Sort of - https://docs.microsoft.com/en-us/dotnet/core/packages#packages lists the "key" root packages. I don't know how stable that list is nor if they're all licensed exactly the same as the .NET Core runtime, so I avoided using it. I think that it's probably best to leave them for now and document that `ignored` can be used to ignore e.g. `runtime.*`, `System.*` if desired.

Reply:
Is there anything to stop someone from building a package like `System.MyPackage`?  Conventions maybe?

Reply:
Purely conventional. Nothing to actually stop someone from doing that AFAIK.